From 4330bea90216e47d2f216cbb03ea9beb09faa54c Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Thu, 21 Feb 2019 12:03:38 +0530
Subject: [PATCH 1021/1051] octeontx2-af: Check LBK link mapped NIXLFs also for
 tx stall

Enabled polling NIXLFs which are transmitting to LBK link
for a possible transmnit stall.

This patch is a continuation to below transmit stall HW issue
workaround patches.
commit c5e4daaaecc7
	("octeontx2-af: Check SQ counters to detect the deadlock")
commit e93af8192a41
	("octeontx2-af: Workaround for NIX transmit stall issue")

Change-Id: Iec35ec78e11be15bca37a0c8b8abce902c4da608
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/4423
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c
index 5bbabb2f6173..2045789f5ec3 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c
@@ -493,9 +493,6 @@ static void rvu_nix_restore_tx(struct rvu *rvu, struct nix_hw *nix_hw,
 	int tl, link;
 
 	link = tx_stall->tl2_link_map[tl2];
-	/* Skip LBK links for now */
-	if (link >= rvu->hw->cgx_links)
-		return;
 
 	tx_stall->stalled_cntr++;
 
@@ -513,8 +510,10 @@ static void rvu_nix_restore_tx(struct rvu *rvu, struct nix_hw *nix_hw,
 	usleep_range(20, 25);
 
 	/* Wait for LMAC TX_IDLE */
-	if (!rvu_cgx_tx_idle(rvu, nix_hw, tl2_txsch, tl2))
-		goto clear_sw_xoff;
+	if (link < rvu->hw->cgx_links) {
+		if (!rvu_cgx_tx_idle(rvu, nix_hw, tl2_txsch, tl2))
+			goto clear_sw_xoff;
+	}
 
 	/* Restore link credits */
 	rvu_wr64(rvu, blkaddr, NIX_AF_TX_LINKX_NORM_CREDIT(link),
-- 
2.17.1

